api/controllers/PostController.js

```diff
@@ -227,6 +227,37 @@ var PostController = {
     .then(() => res.ok({}), res.serverError)
   },
 
+  createFromEmailForm: function (req, res) {
+    try {
+      var tokenData = Email.decodePostCreationToken(req.param('token'))
+    } catch (e) {
+      return res.serverError(new Error('Invalid token: ' + req.param('To')))
+    }
+
+    var attributes = _.merge(
+      {created_from: 'email_form'},
+      _.pick(req.allParams(), ['name', 'description', 'type']))
+
+    var namePrefixes = {
+      'offer': 'I\'d like to share',
+      'request': 'I\'m looking for',
+      'intention': 'I\'d like to create'
+    }
+
+    attributes.name = namePrefixes[attributes.type] + ' ' + attributes.name
+
+    return setupNewPostAttrs(tokenData.userId, attributes)
```

Comment:
this looks like it was copied from elsewhere in PostController -- can you either extract it into a method or make sure that all inapplicable parameters (e.g docs and imageUrl) aren't in the code?